### DISCUSSION TITLE
Laravel Octane and Shared Static Properties between Request Cycles

### DISCUSSION THREAD

**Discussion by rupeshstha** [laravel/framework#47958] [Q&A] [answered]
Question:
I'm currently using Laravel Octane and have encountered an issue related to shared static properties between different request cycles. In my application, I have an 'OrganizationService' class that is called from route > middleware > controller > service. Within this class, there is a function that reads a static property ie already set by middleware to determine the user's organization. However, I noticed that the static property value is being shared between different request cycles, leading to unexpected behavior.

My question is: How does Laravel Octane handle request serving for individual request cycles, and why is the static property value shared across different requests? Is there a recommended approach to avoid such shared state-related issues while using Laravel Octane? Any insights, best practices, or workarounds would be greatly appreciated. Thank you!

**Comment by henzeb** [comment]
You should read https://redberry.international/what-is-octane-in-laravel-things-to-know/

Have you tried storing it in a session?

**Reply by rupeshstha** [reply]
My app is state less. I'm building APIs so i can't save on session.

**Reply by henzeb** [reply]
Ah, then I suggest try and store it in the service container. As, according to the link I gave, this gets cloned for each request.

**Reply by rupeshstha** [reply]
Yeah i could or could not but isn't that gonna be heavy? In my scenario, I not just have to manage organization or the users but the whole concept was to preserve or maintain state of the app and could be manipulate as it goes along with single request cycle. So what does that mean? First, the middleware can sets the user along with necessary data from params or header on "OrganizationState" and the controller > OrganizationService will use the OrganizationState and can manipulate current data with business logic.

This is achieved and maintained with static property and i have managed to destroy the old state by simply removing old state from the middleware itself. I'm just curious to know in depth why is the static property value is shared across different requests? 😄

**Reply by rupeshstha** [reply]
@henzeb Thanks for the reply. I think i got my answer in blog it says those were the thing i should keep track on! 😄 Quite nice explanation. And also is there any guide line to properly setup for production env with octane?

**Reply by rupeshstha** [reply, marked as answer]
I wanted to keep this discussion active, as others might encounter the same issue. To prevent static property data sharing, I used an event to clear the data. In the `octane.php` configuration file, there's a `listeners` array where you can add your own listener under the `RequestReceived::class` key. This event triggers when the application receives a request, allowing us to clear the data at the start of each request cycle.

**Reply by rupeshstha** [reply]
But instead of using the static property and managing states like i have done you can simply use facade design pattern it will make our life easy. 😄
